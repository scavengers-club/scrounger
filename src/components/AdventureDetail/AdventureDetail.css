.adventure-detail {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.adventure-detail img {
  width: 80vw;
  height: 80vw;
  object-fit: cover;
  border-radius: 15px;
  border: 2px solid var(--sand);
  filter: drop-shadow(5px 5px 5px rgb(62, 62, 62));
  margin-bottom: 2vh;
}

.adventure-detail h3 {
  font-family: 'Reenie Beanie', cursive;
  font-size: 200%;
  text-transform: uppercase;
  margin: 3vh;
}

.adventure-detail h4 {
  font-family: 'Reenie Beanie', cursive;
  font-size: 150%;
  text-transform: uppercase;
  margin: 1vh;
}

.adventure-detail button {
  width: 20vw;
  background-color: var(--purple);
  border-radius: 7px;
  padding: 3px;
  margin: 3vw;
}

.adventure-detail p {
  width: 80vw;
  overflow-wrap: break-word;
  text-align: center;
}
