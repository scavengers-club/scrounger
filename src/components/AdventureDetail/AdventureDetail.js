import React from 'react';

//this will be a shared filed, used for displaying info when clicked on home page,
//as well as when a user clicks on their created adventures

export default function AdventureDetail({
  currentUser,
  user_id,
  name,
  image,
  description,
  hint_1,
  hint_2,
  hint_3,
  solution,
  handleEdit,
  handleDelete,
}) {
  return (
    <div>
      <h3>{name}</h3>
      <img width="400px" src={image} />
      <h3>{description}</h3>
      <ol>
        <li>{hint_1}</li>
        <li>{hint_2}</li>
        <li>{hint_3}</li>
      </ol>
      <h3>{solution}</h3>

      {currentUser.id === user_id && (
        <>
          <button onClick={handleEdit}>Edit</button>
          <button onClick={handleDelete}>Delete</button>
        </>
      )}
    </div>
  );
}
