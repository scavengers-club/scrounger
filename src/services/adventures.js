import { client, checkError } from './client';

export async function getAdventures() {
  const resp = await client.from('adventures').select('*');
  return checkError(resp);
}

export async function getAdventureById(id) {
  const resp = await client.from('adventures').select('*').match({ id }).single();
  return checkError(resp);
}

export async function createAdventure(adventure) {
  const resp = await client.from('adventures').insert(adventure);
  return checkError(resp);
}

export async function uploadAdventureImage(adventureId, file) {
  {
    /*
    const ext = file.name.split('.').pop();     //this isolates the file extension 
    await client.storage.from('adv-images').upload(to where?).
  */
  }
}
