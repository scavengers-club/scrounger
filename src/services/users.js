import { client, checkError } from './client.js';

export async function getUser() {
  const session = client.auth.session();
  const { data, error } = await client
    .from('profiles')
    .select('*')
    .match({ id: session.user.id })
    .single();
  if (error) {
    throw error;
  }
  return { ...session.user, ...data };
}

export async function registerUser(email, password, username, avatar) {
  const { user, error } = await client.auth.signUp({ email, password });
  if (error) {
    throw error;
  }
  const resp = await client.from('profiles').insert({ id: user.id, username, avatar }).single();
  if (resp.error) {
    throw error;
  }
  return user;
}

export async function logInUser(email, password) {
  const { user, error } = await client.auth.signIn({ email, password });
  if (error) {
    throw error;
  }
  return user;
}

export async function logOut() {
  const response = await client.auth.signOut();
  return checkError(response);
}
