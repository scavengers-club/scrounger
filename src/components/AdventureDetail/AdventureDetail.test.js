import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import AdventureDetail from './AdventureDetail';
import { getAdventureById } from '../../services/adventures';

test('<AdventureDetail/> should render adventure details', () => {
  const adventure = {
    id: 1,
    name: 'unique adventure',
    description: 'this is a description',
    image:
      'https://media.istockphoto.com/photos/sunset-in-downtown-portland-oregon-picture-id1283364803?b=1&k=20&m=1283364803&s=170667a&w=0&h=CEcB5sIBGs1GX3LhyLxgU53K7rgeP4vNsC4tWhesOAo=',
    hint_1: 'go here',
    hint_2: 'go there',
    hint_3: 'go everywhere',
    solution: 'you are beautiful',
  };
  const { container } = render(
    <MemoryRouter initialEntries={['/adventure/1']}>
      <Route path="/adventure/:id">
        <AdventureDetail
          currentUser={{ id: '575e91dd-2c9c-4297-aa60-6504a62236dc', username: 'violet' }}
          {...adventure}
        />
      </Route>
    </MemoryRouter>
  );

  screen.getByText('Solution: you are beautiful');
  expect(container).toMatchSnapshot();
});
