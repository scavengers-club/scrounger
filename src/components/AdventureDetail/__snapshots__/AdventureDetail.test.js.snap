// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdventureDetail/> should render adventure details 1`] = `
<div>
  <div
    class="adventure-detail"
  >
    <h3>
      unique adventure
    </h3>
    <img
      alt="unique adventure"
      src="https://media.istockphoto.com/photos/sunset-in-downtown-portland-oregon-picture-id1283364803?b=1&k=20&m=1283364803&s=170667a&w=0&h=CEcB5sIBGs1GX3LhyLxgU53K7rgeP4vNsC4tWhesOAo="
    />
    <h4>
      Description:
    </h4>
    <p>
      this is a description
    </p>
    <p>
      Hint 1: 
      go here
    </p>
    <p>
      Hint 2: 
      go there
    </p>
    <p>
      Hint 3: 
      go everywhere
    </p>
    <h4>
      Solution: 
      you are beautiful
    </h4>
  </div>
</div>
`;
